middle_end/flambda2/identifiers/int_ids.ml

```diff
@@ -516,7 +516,9 @@ module Symbol = struct
   let import (data : exported) = Table.add !grand_table_of_symbols data
 
   let map_compilation_unit f (data : exported) : exported =
-    { data with compilation_unit = f data.compilation_unit }
+    Unwrapped_symbol.with_linkage_name
```

Comment:
I should probably rename `with_linkage_name` in that case, since it's a constructor rather than a setter? (I did want to distinguish it from the `for_*` constructors since this isn't wrapping some other thing, it's specifying explicitly the underlying linkage name. Maybe `unsafe_create` with keyword arguments?)

Reply:
Went with `unsafe_create` (no keyword arguments since the `Linkage_name.t` is unambiguous).